pkg/cmd/cli/cmd/tag.go

```diff
@@ -389,6 +389,9 @@ func (o TagOptions) RunTag() error {
 			default:
 				istag.Tag.From.Name = localRef.NameString()
 				istag.Tag.From.Namespace = o.ref.Namespace
+				if o.ref.Namespace == "" {
+					istag.Tag.From.Namespace = o.namespace
```

Comment:
Seems weird to set it once, check the value you just set it to, then set it again. 
```go
var namespace string
if len(o.ref.Namespace) != 0 {
    namespace = o.ref.Namespace
} else {
    namespace = o.namespace
}
istag.Tag.From.Namespace = namespace
```